Feature/ait 1090 ait release 0 4 0 (#3454)
ably/docs: src/pages/docs/ai-transport/api/javascript/core/agent-session.mdx

<MethodSignature>{`connect(): Promise<void>`}</MethodSignature>

> Subscribe to the channel and implicitly attach. The subscribe is unfiltered so channel-history-replayed input events fold into the tree, where each run's input-event watcher can match its trigger. Idempotent: subsequent calls return the same promise. All `AgentRun` methods (`start`, `pipe`, `suspend`, `end`) throw `InvalidArgument` until `connect()` has been called.

Author: This should be simplified to:

```suggestion
Attaches and subscribes to the channel backing the session. Idempotent: subsequent calls return the same promise. All `AgentRun` methods (`start`, `pipe`, `suspend`, `end`) throw `InvalidArgument` until `connect()` has been called.
```

## Create a run <a id="create-run"/>

<MethodSignature>{`createRun(invocation: Invocation, runtime?: RunRuntime<TOutput>): AgentRun<TOutput, TProjection, TMessage>`}</MethodSignature>

> Create a new `AgentRun` from an `Invocation`. Returns synchronously and arms the run's input-event watcher (a passive pre-scan of the tree plus a listener for the trigger's arrival); it publishes nothing to the channel until [`AgentRun.start`](#run-start) is called. The run is registered for cancel routing immediately so early cancels fire the `abortSignal`.

Author: > "Idempotent: subsequent calls return the same promise. All `AgentRun` methods (`start`, `pipe`, `suspend`, `end`) throw `InvalidArgument` until `connect()` has been called."

This is not clear to developers not familiar with SDK internals, and leaks internal implementation detail that is not relevant to the reader. Instead, simply describe that it creates a run for the input event specified in the invocation.

### Returns <a id="create-run-returns"/>

An `AgentRun<TOutput, TProjection, TMessage>` handle for publishing lifecycle events, user messages, and streamed output. See [AgentRun interface](#run) below.

## AgentRun <a id="run"/>

The handle returned by [`createRun`](#create-run). It extends the shared `BaseRun` read-model (`runId`, `status`, `error`, `messages`) with the agent's lifecycle surface.

### Properties

<Table id='RunProperties'>

| Property | Description | Type |
| --- | --- | --- |
| runId | The Run's unique identifier. Known synchronously on the agent (it mints the id for a fresh run, or reads it off the triggering input event for a continuation). | String |
| status | The Run's lifecycle status, read live off the tree. | `RunStatus` |
| error | The terminal error, present exactly when `status` is `'error'`. | `Ably.ErrorInfo` or Undefined |

> | messages | This Run's whole turn: its triggering input message followed by its own streamed output, deduplicated by `codecMessageId`. The unit to persist. See [Hydrate the conversation](#conversation-hydration). | `TMessage[]` |

Author: "turn" is not the correct terminology here; this includes messages across run-suspend boundaries, so a run's messages can span multiple turns for the same run. Apply this fix elsewhere this mistake is made.

| invocationId | The invocation id minted by the agent for this `createRun` call (one per HTTP request). Readable synchronously; the application returns it on the HTTP response. The agent stamps it on every event it publishes for this invocation. | String |
| abortSignal | `AbortSignal` scoped to this Run. Fires when a cancel event arrives. | `AbortSignal` |

> | view | Read-only, leaf-pinned `View<TMessage>` of this Run's branch, from its triggering input back to the conversation root. Drain it with `loadOlder()` for ancestor context to feed the model. Empty until the trigger folds into the tree. See [Hydrate the conversation](#conversation-hydration). | `View<TMessage>` |

Author: This includes too much detail relating to how to perform agent-side hydration; simplify this to just state what the view is, rather than how to use it here.

> | located | Resolves when this Run's triggering input folds into the tree (live or via a `view.loadOlder()` page). [`start`](#run-start) awaits it internally; await it directly only to read the trigger before deciding how to start. | `Promise<void>` |

Author: This leaks implementation detail by describing how the input event is processed (folded into the tree); instead, state that this resolves when the triggering input specified on the run's invocation has been observed on the channel (either via live subscription or through pagination via `loadOlder()`).

</Table>

### Start the run <a id="run-start"/>

<MethodSignature>{`start(): Promise<void>`}</MethodSignature>

> Await [`located`](#run) (the run's input-event watcher resolves the trigger, whether it arrives live or already sits in channel history), then read the trigger's wire headers and publish the opening lifecycle event (`ai-run-start`, or `ai-run-resume` for a continuation). Must be called before `pipe`, `suspend`, or `end`.

Author: Again, "input-event watcher" is not clear to developers unfamiliar with SDK internals; instead describe only the behaviour that matters to the reader; apply this fix elsewhere this pattern is used.

</Table>

## Hydrate the conversation <a id="conversation-hydration"/>

Two accessors expose conversation content, at different scopes:

- `run.messages` is this Run's own turn: its triggering input plus its streamed output. This is the unit to persist, not the value to feed the model in a multi-turn conversation.
- `run.view` is a read-only, leaf-pinned `View` of this Run's full branch, from its triggering input back to the conversation root. This is the value to feed the model.

> To rebuild the prior conversation for the model, drain `run.view` with `loadOlder()` (the sole history driver) for as much ancestor context as you want, then read `getMessages()`:

Author: "the sole history driver" is internal implementation detail that does not need to be surfaced here.
